### ISSUE TITLE
Conical gradient support

### ISSUE THREAD

**Issue by fdnhkj** [bgryszko/react-native-circular-progress#41]
It would be nice to have support for conical gradient.
Has anyone an idea on how to implement it ?

![colorgradient](https://cloud.githubusercontent.com/assets/5569608/19495073/20b71cd4-9582-11e6-9830-02d2ce2f2b47.jpg)

**Comment by GroupXTech** [comment]
Any luck on finding your conical gradient... I need that in a slider as well

**Comment by morelazers** [comment]
Yes I'm looking for something similar, I want to animate the colour of the fill.

I can get it to work by removing the validation of `PropTypes.string` but it is incredibly slow.

**Comment by bgryszko** [comment]
@fdnhkj Cannonical gradients aren't supported in SVG. In my other component (https://github.com/bgryszko/react-native-circular-slider) I achieve exactly what you're looking for by splitting circle into segments and draw small linear gradients – in practice if gradient colors difference is subtle or there are enough "segments" it'll look as if it's cannonical gradient. 

Link to specific code:
https://github.com/bgryszko/react-native-circular-slider/blob/master/src/CircularSlider.js#L161-L168

**Comment by fdnhkj** [comment]
Indeed, I forgot to update this issue with my findings, I'll try to find back my repo & share it.

Basically I switched to react-native-svg and as you mentioned splited the circle in multiple segments with attached linear gradients (see [http://jsfiddle.net/Cy7ec/3/](http://jsfiddle.net/Cy7ec/3/)).

The conical gradient was looking fine on both iOS & Android but I had quite some issues with the Android animations (lots of dropped fps).

It's maybe worth trying with the latest versions of react-native & react-native-svg.
I will give you updates ASAP.

**Comment by bgryszko** [comment]
That's great! In fact this library would need switch over to react-native-svg as well.

Bart

**Comment by fdnhkj** [comment]
Voilà, I published my findings after upgrading to latest `react-native` and `react-native-svg`  in [fdnhkj/react-native-conical-gradient](https://github.com/fdnhkj/react-native-conical-gradient).

It looks like there are some dropped fps. I don't really see tho how to improve it tho, suggestions are welcome 🙂

Fidan

**Comment by bgryszko** [comment]
I'm happy to accept PR if anybody would be able to merge this into main lib.

**Comment by zoharlevin** [comment]
+1

**Comment by t2krew** [comment]
+1

**Comment by rodrigooler** [comment]
+1

**Comment by valerybodak** [comment]
@fdnhkj 
Have you implemented your conical gradient? Is yes, can you share your code please. Thanks a lot!

**Comment by NeXTs** [comment]
@fdnhkj based on [this research](https://github.com/kdzwinel/progress-bar-animation) the only possible solution to improve FPS is using canvas instead of svg

**Comment by uppercase-army** [comment]
+1

**Comment by hanishcodebrew** [comment]
@fdnhkj How to make it spin in the opposite direction?

**Comment by MikePodgorniy** [comment]
@hanishcodebrew you can wrap component into
```javascript
<View style={ { transform: [ { scaleX: -1 }] } }>
``` 
or use similar code
https://github.com/MikePodgorniy/react-native-conical-gradient-progress/blob/de35329780ed6431ff21710dc8a6c69b639263e8/CircularProgress.js#L175-L183

**Comment by hanishcodebrew** [comment]
Thanks @MikePodgorniy for the instant reply. It works. I am facing a issue in Android.
![screen shot 2018-12-05 at 1 14 52 pm](https://user-images.githubusercontent.com/40761838/49498083-f806b100-f88f-11e8-9bb9-fbeda53a0f65.png)
![screen shot 2018-12-05 at 1 14 52 pm](https://user-images.githubusercontent.com/40761838/49498090-fb9a3800-f88f-11e8-997d-a921219a2862.png)

I have given 45 to fill but its showing complete circle. And also can we add image and text inside the circle ?

**Comment by bartgryszko** [comment]
@MikePodgorniy I'm not actively participating in developing this library as I'm not working on any RN projects currently. That said, feel free to add PR, if not me, maybe @Jacse or @markusl would be able to review and merge it.

**Comment by MikePodgorniy** [comment]
@hanishcodebrew yes, you can 👍 use render callback to this. but you need to rescale your inner components again
https://github.com/MikePodgorniy/react-native-conical-gradient-progress/blob/de35329780ed6431ff21710dc8a6c69b639263e8/example/App.js#L25-L32

**Comment by Angelk90** [comment]
@fdnhkj,@GroupXTech,@morelazers,@bartgryszko,@fdnhkj,@NeXTs,@MikePodgorniy,@hanishcodebrew :
I wanted to know how it ended, it also served me.

@fdnhkj: Your module react-native-conical-gradient, I do not find it on npm you can upload it, can you also put an example code?

@MikePodgorniy: Your module react-native-conical-gradient-progress, I do not find it on npm you can upload it, can you also put an example code?

For both:
The sample code of the images of your modules would be a step forward.

**Comment by MikePodgorniy** [comment]
@Angelk90 Try now, it is available as an npm-package:
https://www.npmjs.com/package/react-native-conical-gradient-progress.

You can find an example here – https://github.com/MikePodgorniy/react-native-conical-gradient-progress/tree/master/example

if you use clear react-native you should install react-native-svg.

**Comment by Angelk90** [comment]
@MikePodgorniy :  I do not understand why among the dependencies you did not put svg in the package.json?

**Comment by Angelk90** [comment]
@MikePodgorniy : I tried to convert your module to try it on https://snack.expo.io/
I am given the following error:
`TypeError: undefined is not an object (evaluating 'o.map')
`
Line 27 of CircularProgress:
Here is the link where you can try it: https://snack.expo.io/HkDheSJmE

**Comment by Angelk90** [comment]
@MikePodgorniy : Do not allow me to create issues in the indicated project.

**Comment by MikePodgorniy** [comment]
@Angelk90  sorry, try again

**Comment by ngnclht1102** [comment]
> Voilà, I published my findings after upgrading to latest `react-native` and `react-native-svg` in [fdnhkj/react-native-conical-gradient](https://github.com/fdnhkj/react-native-conical-gradient).
> 
> It looks like there are some dropped fps. I don't really see how to improve it tho, suggestions are welcome 🙂
> 
> ![](https://raw.githubusercontent.com/fdnhkj/react-native-conical-gradient/master/conical-gradient-ios.gif)
> ![](https://raw.githubusercontent.com/fdnhkj/react-native-conical-gradient/master/conical-gradient-android.gif)
> 
> Fidan

I have the same problem as well, the UI's fps is dropped to 2x. Any ideas on how to improve it?

**Comment by thanakij** [comment]
I have been doing something similar. My approach uses "LinearGradient" react-native-svg.

You can give it a try by:

`npm install react-native-circular-progress-gradient@1.3.8`

You will need to add **tintColorSecondary** prop to trigger the gradient mode.

![](https://i.imgur.com/zm4b8u5.jpeg)

**Comment by minafa96** [comment]
> I have been doing something similar. My approach uses "LinearGradient" react-native-svg.
> 
> You can give it a try by:
> 
> `npm install react-native-circular-progress-gradient@1.3.8`
> 
> You will need to add **tintColorSecondary** prop to trigger the gradient mode.
> 
> ![](https://camo.githubusercontent.com/4ae828ca94bee88e5a616c0cad6e050d06f335f9112690e07d5721b91d456ea4/68747470733a2f2f692e696d6775722e636f6d2f7a6d34623875352e6a706567)

GREAT, THANK YOU! 👍
